src/strategies/samurailegends-smg-staking-balance/index.ts

```diff
@@ -0,0 +1,33 @@
+import { BigNumberish } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+import { Multicaller } from '../../utils';
+
+export const author = 'Samurai-Legends';
+export const version = '0.1.0';
+
+const abi = ['function userStake(address account) public view returns (uint)'];
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const multi = new Multicaller(network, provider, abi, { blockTag });
+  addresses.forEach((address) =>
+    multi.call(address, options.address, 'userStake', [address])
+  );
+
+  const result: Record<string, BigNumberish> = await multi.execute();
+
+  return Object.fromEntries(
+    Object.entries(result).map(([address, balance]) => [
+      address,
+      parseFloat(formatUnits(balance, options.decimals))
```

Comment:
Hi @ThijmenMaus I think you can use https://snapshot.org/#/strategy/contract-call for this